test/helper.lua

```diff
@@ -3,101 +3,219 @@ local fio = require("fio")
-        vinyl:create_index("multikey_index",
-                {type = "TREE", parts = { {7, type = "str", path = "data[*].name", is_nullable = true} }})
+        space:create_index("json_path_index", {
+            type = "TREE",
+            parts = {
+                {
+                    field = 6,
+                    type = "scalar",
+                    path = "age",
+                    is_nullable = true
+                }
+            }
+        })
+        space:create_index("multikey_index", {
+            type = "TREE",
+            parts = {
+                {
+                    field = 7,
+                    type = "str",
+                    path = "data[*].name"
+                }
+            }
+        })
+        if engine ~= "vinyl" then
+            space:create_index("functional_index", {
+                type = "TREE",
+                parts = {
+                    {
+                        field = 1,
+                        type = "string"
+                    }
+                },
+                func = "tree_func"
+            })
+        end
     end
 
-    local bitset = box.schema.create_space("bitset", { if_not_exists = true })
-    bitset:create_index("primary", {type = "TREE", parts={ 1 }})
-    bitset:create_index("index_for_first_name",
-            {type = "BITSET", parts={ {field = 2, type = "string"} }, unique = false})
-end)
+    return space
+end
+
+function helpers.create_space_with_hash_index(engine)
+    local space = create_space("hash", engine)
+    space:create_index("primary", {
+        type = "HASH",
+        parts = {
+            {
+                field = 1
+            }
+        }
+    })
+    space:create_index("index_for_first_name", {
+        type = "HASH",
+        parts = {
+            {
+                field = 2
+            }
+        }
+    })
+    space:create_index("multipart_index", {
+        type = "HASH",
+        parts = {
+            {
+                field = 1
+            },
+            {
+                field = 2
+            }
+         }
+     })
+
+    return space
+end
+
+function helpers.create_space_with_bitset_index(engine)
+    local space = create_space("bitset", engine)
+    space:create_index("primary", {
+        type = "TREE",
+        parts = {
+            {
+                field = 1
+            }
+        }
+    })
+    space:create_index("index_for_first_name", {
+        type = "BITSET",
+        parts = {
+            {
+                field = 2,
+                type = "string"
+            }
+        },
+        unique = false
+    })
+
+    return space
+end
 
 t.after_suite(function()
     fio.rmtree(t.datadir)
 end)
 
-function helpers.init_spaces(g)
-    g.tree = box.space.tree
-    g.hash = box.space.hash
-    g.vinyl = box.space.vinyl
-    g.bitset = box.space.bitset
-end
+t.before_suite(function()
+    t.datadir = fio.tempdir()
+    box.cfg{
```

Comment:
Just a comment. Should we test memtx against box.cfg.memtx_use_mvcc_engine?

Reply:
I don't see any reasons why we need to test with option `memtx_use_mvcc_engine`.
But if we have such reasons it should be done in scope of another ticket.
This PR about tests refactoring to add parameterizing.